src/android/Mapbox.java

```diff
@@ -168,7 +183,23 @@ public void run() {
             FrameLayout.LayoutParams params = new FrameLayout.LayoutParams(webViewWidth - left - right, webViewHeight - top - bottom);
             params.setMargins(left, top, right, bottom);
             mapView.setLayoutParams(params);
-
+			//@anothar we have to override back button or app will just close
+            mapView.setOnKeyListener(new View.OnKeyListener() {
```

Comment:
@cusspvz 
1) Event is not delegated to parent container of mapview and webview. Thus your app doesn't  force close.
2) Event is delegated to webview. WebView invokes js events to backbutton. Thus all cordova handling code for backbutton works correctly.
This problem exists because mapview is not part of webview. It's separate control. It won't delegate it events to js code because id knows nothing about js. That is why we have to do it manually or if your user will press hardware backbuttom while map is focused your app will unexpectedly shutdown(unexpectedly only for cordova).